_episodes/09-branches.md

```diff
@@ -1,79 +1,61 @@
 ---
 layout: episode
-title: Branching and Merging
+title: Branching and merging
 teaching: 20
 exercises: 15
 questions:
   - How can I or my team work on multiple features in parallel?
   - How to combine the changes of parallel tracks of work?
-  - What is branching and when should I do it?
   - How can I permanently reference a point in history, like a software
     version?
 objectives:
   - Be able to create and merge branches.
   - Know the difference between a branch and a tag.
 keypoints:
   - A branch is a division unit of work, to be merged with other units of work.
-  - Creating branches is simple in Git.
-  - Merging branches is typically straightforward.
   - A tag is a pointer to a moment in the history of a project.
 ---
 
-
-## Introduction
-In the previous section we made guacamole Git style. Our procedure was serial
-(which is a cool way to say doing one thing after another in a sequence). This
-sequence of events are depicted in the diagram below. At this point we shall
-learn few more concepts and terms. The sequence of events depicted in the
-diagram is called a “branch”. When there is only one branch and if we do not
-rename it, Git assigns the name “master” to it. The latest state of the branch
-is referred as the “HEAD”. The term HEAD is better explained with the reference
-to the recording head of tape recorders (if you are too young to know what
-this is ask around). So if someone presses the record button it will start the
-recording from where the HEAD is.  At any point of operation a Git repository
-can have only one HEAD.
-
-![Master branch]({{ site.baseurl }}/img/master_branch.svg 
-"master branch"){:class="img-responsive" style="max-width:60%"}
-
----
-
 ## Motivation for branches
 
+In the previous section we tracked a guacamole recipe with Git.
+
 Up until now our repository had only one branch with one commit coming
 after the other:
 
 ![Linear]({{ site.baseurl }}/img/gitink/git-branch-1.svg "Linear git
 repository"){:class="img-responsive"}
 
-However, software development is often not linear:
+- Commits are depicted as little boxes with abbreviated hashes.
+- The sequence of commits forms a **branch**.
+- Here the branch is called "master".
+- "HEAD" is the current position (remember the recording head of tape recorders?).
 
-  - Maintenance of releases, bug-fixes, and patches.
-  - Collaborative work (everybody would be exposed to your bugs; you would be
-    exposed to bugs of other people).
-  - We do not have time to write perfect code immediately, it would be nice to
-    have the possibility to experiment somewhere aside.
-  - Interrupted work (we typically work on several longer term projects at the
-    same time).
+Software development is often not linear:
```

Comment:
I don't know how (too late), but I have a feeling this could be motivated shorter.  These are all good reasons, but...

I was recently thinking of comparison to wikis: it's OK if a wiki is in a bad state, someone will fix later.  But if you need master to always be good, you will need to be able to separate different lines of work really well.  I'm not saying this is the best comparison here, just saying.

Reply:
OK I will try to shorten it tomorrow.